### PR TITLE
Adding Testing Library support for Cypress

### PR DESCRIPTION
**Pull request by weerd**
### What's this PR do?

This pull request adds support for using [Testing Library](https://testing-library.com/docs/cypress-testing-library/intro) utilities in our Cypress tests!

It also updates the `StartVoterRegistrationForm.js` to use the new `data-testid` attribute for testing to then allow us to use the `findByTestId()` helper function in our Cypress tests.

Some of these changes were made in #2155 but they were merged into a non-master branch 😬 

### How should this be reviewed?

👀

### CODE REVIEW

**Comment by weerd** [comment]
cc @Dinnall

File: cypress/integration/beta-voter-registration-drive-page.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ describe('Beta Voter Registration Drive (OVRD) Page', () => {
     });
   });
 
+  /** @test */
```
Comment (mendelB) [inline cypress/integration/beta-voter-registration-drive-page.js:46]: Are the `@test` annotations a new convention we want to introduce? I figure maybe worth throwing into the communal docs, or where might the best place be?

**Comment by Dinnall** [comment]
> cc @Dinnall

🤯 I just remembered I didn't merge that branch into master. I should have just made another commit in the same branch.

File: cypress/integration/beta-voter-registration-drive-page.js (hunk 1 above)
Comment (weerd) [reply cypress/integration/beta-voter-registration-drive-page.js:46]: oh ya, I was trying it out related to this [conversation](https://github.com/DoSomething/phoenix-next/pull/2132#discussion_r425995637).

I was planning on updating the Communal Docs after this now that the Testing Library helper can be used with Cypress :)

**Comment by weerd** [comment]
> 🤯 I just remembered I didn't merge that branch into master. I should have just made another commit in the same branch.

No worries! It should be updated now after merging this PR so just pull down and you should be all set 👯

File: cypress/integration/beta-voter-registration-drive-page.js (hunk 1 above)
Comment (mendelB) [reply cypress/integration/beta-voter-registration-drive-page.js:46]: Ah the ole Diego breathing room ™️ :) Thanks for that context link!

**Pull request merged**